### PR TITLE
#157050547 Fix display of buildings by click of country buttons

### PR DESCRIPTION
**Pull request by ivanatu**
#### What does this PR do?

This PR fixes the issue of displaying the correct buildings for a particular country. 

#### Description of Task to be completed?

This task meant creation of the building adapter and the country adapter for easy access of the data we collect from the API. Plus some of the display features had to be modified as to look good as per material designs. 

#### How should this be manually tested?

This can be manually tested by cloning the app into your local android studio and running using any emulator of your choice

#### Any background context you want to provide?

This PR combines 2 features , that creation of both the building adapter and the country adapter.

#### What are the relevant pivotal tracker stories?

[157050547](https://www.pivotaltracker.com/story/show/157050547)

#### Screenshots (if appropriate)

may-21-2018 18-19-48

#### Questions:

None

### CODE REVIEW

**Comment by TheDancerCodes** [comment]
@ivanatu 
I am thinking, from a UX perspective, wouldn't it be better to display something to the users when they click on a country button and there's no data available? 
I recommend this instead of them interacting with a blank screen as evidenced in the GIF you shared in your PR.
cc: @chykehyman

File: app/src/main/java/com/andela/mrm/room_booking/building/BuildingFragment.java [hunk 1]
```diff
@@ -66,8 +66,7 @@ public void displayBuildings(final List<AllLocationsQuery.Block> mBlock) {
         mRecyclerView.setHasFixedSize(true);
         if (numCols != 0) {
             mLayoutManager = new GridLayoutManager(getContext(), numCols);
-        }
-        else {
+        } else {
```
Comment (TheDancerCodes) [inline app/src/main/java/com/andela/mrm/room_booking/building/BuildingFragment.java]: @ivanatu 
Is this where we handle the case of no results being returned from the click?
cc: @chykehyman

**Comment by TheDancerCodes** [comment]
@chykehyman 
Thanks for your response.

I am aware that one of the floors in the Lagos doesn't have meeting rooms per say, it is just an open space. I think this is the bootcamp space.

What happens to such an edge case? 
Will the user just see a blank screen?
cc: @ivanatu

File: app/src/main/java/com/andela/mrm/room_booking/building/BuildingFragment.java (hunk 1 above)
Comment (ivanatu) [reply app/src/main/java/com/andela/mrm/room_booking/building/BuildingFragment.java]: yes, if that condition is not added , the app crashes everytime you click on a country with no buildings. but like @chykehyman  said , every country should atleast return 1 building. and hence moving forward when we use the backend API, that check will be useless and should be removed.

**Comment by ivanatu** [comment]
Am thinking if the floor has no meeting room, then that floor shouldn't appear on the floors for that building. Back end should be able to cater for this.

**Comment by TheDancerCodes** [comment]
Ah I see your points of view gentlemen. @ivanatu @chykehyman 
Thank you for this very constructive discourse.

That being said the PR LGTM.
I will proceed to merge it.

**Pull request merged**